File: src/main/java/upbrella/be/user/dto/response/SingleBlackListResponse.java
package upbrella.be.user.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import lombok.Getter;
import upbrella.be.user.entity.BlackList;

import java.time.LocalDateTime;

@Getter
@Builder
public class SingleBlackListResponse {

    private long id;
    @JsonFormat(shape= JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd kk:mm:ss")
    private LocalDateTime blockedAt;

    public static SingleBlackListResponse of(BlackList blackList) {

        return SingleBlackListResponse.builder()
                .id(blackList.getId())
                .blockedAt(blackList.getBlockedAt())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/user/dto/response/SingleBlackListResponse.kt
package upbrella.be.user.dto.response

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime
import java.time.ZoneId
import upbrella.be.user.entity.BlackList

data class SingleBlackListResponse(
    val id: Long,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    val blockedAt: LocalDateTime
) {
    companion object {

        private val KST = ZoneId.of("Asia/Seoul")

        private fun toKst(time: LocalDateTime): LocalDateTime {
            return time.atZone(ZoneId.of("UTC")).withZoneSameInstant(KST).toLocalDateTime()
        }

        fun of(blackList: BlackList): SingleBlackListResponse {
            return SingleBlackListResponse(
                id = blackList.id!!,
                blockedAt = toKst(blackList.blockedAt)
            )
        }
    }
}
